src/stdafx.h

```diff
@@ -81,29 +82,62 @@
 
 #include "zip.h"
 
+namespace fs = std::tr2::sys;
 #else
+#include "boost/filesystem.hpp"
+#include <stdarg.h>
+#include <unistd.h>
+namespace fs = boost::filesystem;
 
-//#include <unistd.h>
+#include <sys/syscall.h>
 
+#ifdef SYS_gettid
+pid_t gettid() { return syscall(SYS_gettid); }
+#else
+#error "SYS_gettid unavailable on this system"
 #endif
 
-namespace fs = std::tr2::sys;
+#endif
 
 #if defined(_MSC_VER)
 #define RHOST_EXPORT __declspec(dllexport)
 #define RHOST_IMPORT __declspec(dllimport)
+#define RHOST_NORETURN __declspec(noreturn)
+#define RHOST_NOTHROW __declspec(nothrow)
```

Comment:
I think we can just start using `noexcept` instead. All the compilers we care about should support it. The `SCOPE_WARDEN` code was from older times.